vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import os
+import pathlib
+import re
+import sys
+from typing import List, Optional
+
+from vaccine_feed_ingest_schema import location as schema
+
+from vaccine_feed_ingest.utils.log import getLogger
+
+logger = getLogger(__file__)
+
+
+SOURCE_NAME = "us_giscorps_vaccine_providers"
+
+
+def _get_availability(site: dict) -> schema.Availability:
+    appt_only = site["attributes"]["appt_only"]
+    if appt_only == "Yes":
+        return schema.Availability(appointments=True)
+    elif appt_only is not None:
+        return schema.Availability(drop_in=True)
+
+    return None
+
+
+def _get_id(site: dict) -> str:
+    data_id = site["attributes"]["GlobalID"]
+
+    # Could parse these from the input file name, but do not for now to avoid
+    # accidental mutation.
+    arcgis = "c50a1a352e944a66aed98e61952051ef"
+    layer = 0
+
+    return f"{arcgis}_{layer}_{data_id}"
+
+
+def _get_inventory(site: dict) -> Optional[List[schema.Vaccine]]:
+    # vaccines_field = site["attributes"]["flu_vaccinations"].lower().split(",")
+
+    # potentials = {
+    #     "pfizer": schema.Vaccine(vaccine="pfizer"),
+    #     "moderna": schema.Vaccine(vaccine="moderna"),
+    #     "janssen": schema.Vaccine(vaccine="janssen"),
+    #     "jjj": schema.Vaccine(vaccine="janssen"),
+    # }
+
+    # inventory = []
+
+    # for vf in vaccines_field:
+    #     try:
+    #         inventory.append(potentials[vf])
+    #     except KeyError as e:
+    #         logger.error("Unexpected vaccine type: %s", e)
+
+    # if len(inventory) > 0:
+    #     return inventory
+
+    return None
+
+
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+    if site["attributes"]["phone"]:
+        sourcePhone = re.sub("[^0-9]", "", site["attributes"]["phone"])
+        if len(sourcePhone) == 11:
+            sourcePhone = sourcePhone[1:]
+
+        # TODO: handle 3-digit phone numbers like 211, 411 .etc
+        if len(sourcePhone) == 10:
+            phone = f"({sourcePhone[0:3]}) {sourcePhone[3:6]}-{sourcePhone[6:]}"
+            contacts.append(schema.Contact(phone=phone))
+
+    # if site["attributes"]["publicEmail"]:
+    #     contacts.append(schema.Contact(email=site["attributes"]["publicEmail"]))
+
+    # there are multiple urls, vaccine, agency, health dept. etc
+    if site["attributes"]["vaccine_url"]:
+        url = site["attributes"]["vaccine_url"]
+        url = sanitize_url(url)
+        if url:
+            contacts.append(schema.Contact(website=url))
+
+    if len(contacts) > 0:
+        return contacts
+
+    return None
+
+
+def sanitize_url(url):
+    url = url.strip()
+    url = url.replace("#", "")
+    url = url.replace("\\", "/")  # thanks windows
```

Comment:
😩

Reply:
yeah, this isnt the best but there were a couple URLS in the dataset that were causing problems (as i mentioned in the discord)

one is literally a physical real-world address (that gets caught by the check for spaces in the url below this), one ended in a # and for some reason that broke the validation, and then there was the one with backslashes in it